sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/TransientIOErrorsRetryingIterator.scala

```diff
@@ -112,7 +120,31 @@ private class TransientIOErrorsRetryingIterator[TSparkRow]
           currentFeedResponseIterator.get
       }
 
-      if (feedResponseIterator.hasNext) {
+      val hasNext: Boolean = try {
+        Await.result(
+          Future {
+            feedResponseIterator.hasNext
+          }(TransientIOErrorsRetryingIterator.executionContext),
+          maxPageRetrievalTimeout)
+      } catch {
+        case timeoutException: TimeoutException =>
```

Comment:
if the error(OperationCancelledException) comes from the underlying query, should we log the diagnostics from it?